ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/offset/LatestByOffset.java

```diff
@@ -124,5 +161,50 @@ public T map(final Struct agg) {
       }
     };
   }
+  
+  @UdafFactory(description = "Latest N by offset")
+  static <T> Udaf<T, List<Struct>, List<T>> latestN(final Schema structSchema, final int topNSize) {
+    return new Udaf<T, List<Struct>, List<T>>() {
+      @Override
+      public List<Struct> initialize() {
+        return new ArrayList<Struct>();
+      }
+
+      @Override
+      public List<Struct> aggregate(final T current, final List<Struct> aggregate) {
+        if (current == null) {
+          return aggregate;
+        }
+
+        aggregate.add(createStruct(structSchema, current));
+        final int currentSize = aggregate.size();
+        if (currentSize <= topNSize) {
+          return aggregate;
+        } else {
+          return aggregate.subList(currentSize - topNSize, currentSize);
+        }
+      }
+
+      @Override
+      public List<Struct> merge(final List<Struct> aggOne, final List<Struct> aggTwo) {
+        final List<Struct> merged = new ArrayList<>();
+        merged.addAll(aggOne);
+        merged.addAll(aggTwo);
+        final int currentSize = merged.size();
+        Collections.sort(merged, structComparator);
+        if (currentSize <= topNSize) {
+          return merged;
+        } else {
+          return merged.subList(currentSize - topNSize, currentSize);
+        }
+      }
+
+      @Override
+      @SuppressWarnings("unchecked")
+      public List<T> map(final List<Struct> agg) {
+        return (List<T>) agg.stream().map(s -> s.get(VAL_FIELD)).collect(Collectors.toList());
+      }
+    };
+  }
```

Comment:
Please make corresponding changes to this code, as were suggested for the `EarliestByOffset` code.